src/auto-evo/simulation/MichePopulation.cs

```diff
@@ -141,157 +161,124 @@ private static List<Species> CopyInitialPopulationsToResults(SimulationConfigura
-        IEnumerable<Species> genericSpecies, Random random, SimulationCache cache,
-        IAutoEvoConfiguration autoEvoConfiguration, WorldGenerationSettings worldSettings)
+        IEnumerable<Species> genericSpecies, Random random, SimulationCache cache)
     {
         _ = random;
 
         var populations = simulationConfiguration.Results;
         bool trackEnergy = simulationConfiguration.CollectEnergyInformation;
-        bool dayNightCycle = worldSettings.DayNightCycleEnabled;
-
-        // This algorithm version is for microbe species
-        // TODO: add simulation for multicellular
-        var species = genericSpecies.Select(s => s as MicrobeSpecies).Where(s => s != null).Select(s => s!)
-            .ToList();
 
-        // Skip if there aren't any species in this patch
-        if (species.Count < 1)
-            return;
+        var miche = populations.GetMicheForPatch(patch);
 
-        var energyBySpecies = new Dictionary<MicrobeSpecies, float>();
-        foreach (var currentSpecies in species)
-        {
-            energyBySpecies[currentSpecies] = 0.0f;
-        }
+        foreach (var extraSpecies in simulationConfiguration.ExtraSpecies)
+            miche.InsertSpecies(extraSpecies, patch, cache);
 
-        bool strictCompetition = autoEvoConfiguration.StrictNicheCompetition;
+        // This prevents duplicates caused by ExtraSpecies
+        var species = new HashSet<Species>();
 
-        var niches = new List<FoodSource>
+        foreach (var currentSpecies in genericSpecies)
         {
-            new EnvironmentalFoodSource(patch, Sunlight, Constants.AUTO_EVO_SUNLIGHT_ENERGY_AMOUNT),
-            new EnvironmentalFoodSource(patch, Temperature, Constants.AUTO_EVO_THERMOSYNTHESIS_ENERGY_AMOUNT),
-            new CompoundFoodSource(patch, Glucose, dayNightCycle),
-            new CompoundFoodSource(patch, HydrogenSulfide, dayNightCycle),
-            new CompoundFoodSource(patch, Iron, dayNightCycle),
-            new ChunkFoodSource(patch, "marineSnow"),
-            new ChunkFoodSource(patch, "ironSmallChunk"),
-            new ChunkFoodSource(patch, "ironBigChunk"),
-        };
+            if (simulationConfiguration.ExcludedSpecies.Contains(currentSpecies))
+                continue;
 
-        foreach (var currentSpecies in species)
+            species.Add(currentSpecies);
+        }
+
+        foreach (var currentSpecies in simulationConfiguration.ExtraSpecies)
         {
-            niches.Add(new HeterotrophicFoodSource(patch, currentSpecies, cache));
+            species.Add(currentSpecies);
         }
 
-        foreach (var niche in niches)
+        // Skip if there aren't any species in this patch
+        if (species.Count < 1)
+            return;
+
+        var leafNodes = new List<Miche>();
+        miche.GetLeafNodes(leafNodes, x => x.Occupant != null);
+
+        var energyDictionary = species.ToDictionary(x => x, _ => 0.0);
```

Comment:
Was this intentionally or accidentally using doubles? I'll swap this to floats.

Reply:
Or actually I'll just for safety put these back to double as if the numbers get so high that it is important for accuracy.

Reply:
Accidental, it should have been a float

Reply:
Alright, I'll put it as floats then but add a comment about checking it.